### ISSUE TITLE
Using showError after useAsyncData does not prevent page from loading, kinda

### ISSUE THREAD

**Issue by henk23** [nuxt/nuxt#28557]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.12.4
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      compatibilityDate, devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-vickjg?file=pages%2Findex.vue

### Describe the bug

When showing an error page after useAsyncData() fails, the behavior is weird.
Some further code of <script setup> is executed, but not all of it.
And a component that receives a required prop complains about it being undefined.

Expectation would be:
- As soon as showError() is invoked, all further code execution stops.
- Components should not be rendered at all, so should not warn about missing props, since we show the error page and not the page that failed loading.

Or am I completely misunderstanding something here or doing it wrong?

**Comment by Serpentarius13** [comment]
I think Language Server can't know if script execution would be paused or not. At docs it is said: `It is recommended instead to use throw createError().`. I think that's because showError is not stopping code execution at the time it invoked - the implementation changes useError's Proxy value to a newly created error and the state of ui is updated right at accessing nearest ref (try to remove accessing data.value or add something before). `throw`ing error would make top application listener to catch it immediately instead and prevent any further execution

**Comment by henk23** [comment]
Yes, fair enough, replace `showError()` with `throw createError()` as the docs say.

The child component will still throw a prop warning:

```
[Vue warn]: Invalid prop: type check failed for prop "title". Expected String with value "undefined", got Undefined 
```

So the child components of that page are still are trying to be set-up or created.

This should be prevented, right?

**Comment by ptdev** [comment]
Hi,

I've recently came across what I believe is the same issue as reported here.

So, for example, if I have this page (simplified):


```vue
<script setup lang="ts">
const route = useRoute()
const content = await useProductData(`/tours/${route.params.prod}`)
if (!content) {
    throw createError({ statusCode: 404 })
}
</script>

<template>
    <div>
        <SomeComponent :content="content" />
    </div>
</template>
```


And for some reason the content var is empty (usually by accessing a non-existing route), then although we do get a 404 error page, the console shows the following error:

`[Vue warn]: Invalid prop: type check failed for prop "content". Expected Object, got Undefined (...)`

Which means that execution appears to continue after throwing createError().

Is there anyway to prevent this and stop everything once an error is thrown?
